newIDE/app/src/KeyboardShortcuts/index.js

```diff
@@ -19,6 +19,69 @@ type KeyType =
   | 'numrow-arith' // Numrow-, Numrow=
   | 'other'; // Tab, Space
 
+const eventWhichToCode = {
+  '48': 'Digit0',
+  '49': 'Digit1',
+  '50': 'Digit2',
+  '51': 'Digit3',
+  '52': 'Digit4',
+  '53': 'Digit5',
+  '54': 'Digit6',
+  '55': 'Digit7',
+  '56': 'Digit8',
+  '57': 'Digit9',
+  '65': 'KeyA',
+  '66': 'KeyB',
+  '67': 'KeyC',
+  '68': 'KeyD',
+  '69': 'KeyE',
+  '70': 'KeyF',
+  '71': 'KeyG',
+  '72': 'KeyH',
+  '73': 'KeyI',
+  '74': 'KeyJ',
+  '75': 'KeyK',
+  '76': 'KeyL',
+  '77': 'KeyM',
+  '78': 'KeyN',
+  '79': 'KeyO',
+  '80': 'KeyP',
+  '81': 'KeyQ',
+  '82': 'KeyR',
+  '83': 'KeyS',
+  '84': 'KeyT',
+  '85': 'KeyU',
+  '86': 'KeyV',
+  '87': 'KeyW',
+  '88': 'KeyX',
+  '89': 'KeyY',
+  '90': 'KeyZ',
+  '112': 'F1',
+  '113': 'F2',
+  '114': 'F3',
+  '115': 'F4',
+  '116': 'F5',
+  '117': 'F6',
+  '118': 'F7',
+  '119': 'F8',
+  '120': 'F9',
+  '121': 'F10',
+  '122': 'F11',
+  '123': 'F12',
+  '188': 'Comma',
+};
+
+const getCodeFromEvent = (e: KeyboardEvent): string => {
+  if (
+    (e.which >= 65 && e.which <= 90) ||
+    (e.which >= 48 && e.which <= 57) ||
+    (e.which >= 112 && e.which <= 123) ||
+    e.which === 188
```

Comment:
could this be simplified by `if (e.which in eventWhichToCode)` ?
From my tests, this seems to work